crates/rustc_codegen_spirv/src/codegen_cx/entry.rs

```diff
@@ -104,6 +105,152 @@ impl<'tcx> CodegenCx<'tcx> {
             });
     }
 
+    pub fn lazy_add_bindless_descriptor_sets(&self) {
+        self.bindless_descriptor_sets
+            .replace(Some(BindlessDescriptorSets {
+                // all storage buffers are compatible and go in set 0
+                buffers: self.buffer_descriptor_set(0),
+
+                // sampled images are all compatible in vulkan, so we can overlap them
+                sampled_image_1d: self.texture_bindless_descriptor_set(
+                    1,
+                    rspirv::spirv::Dim::Dim1D,
+                    true,
+                ),
+                sampled_image_2d: self.texture_bindless_descriptor_set(
+                    1,
+                    rspirv::spirv::Dim::Dim2D,
+                    true,
+                ),
+                sampled_image_3d: self.texture_bindless_descriptor_set(
+                    1,
+                    rspirv::spirv::Dim::Dim3D,
+                    true,
+                ),
+                // jb-todo: storage images are all compatible so they can live in the same descriptor set too
+            }));
+    }
+
+    fn buffer_descriptor_set(&self, descriptor_set: u32) -> Word {
+        let uint_ty = SpirvType::Integer(32, false).def(rustc_span::DUMMY_SP, self);
+
+        let runtime_array_uint =
+            SpirvType::RuntimeArray { element: uint_ty }.def(rustc_span::DUMMY_SP, self);
+
+        let buffer_struct = SpirvType::Adt {
+            def_id: None,
+            size: Some(Size::from_bytes(4)),
+            align: Align::from_bytes(4).unwrap(),
+            field_types: vec![runtime_array_uint],
+            field_offsets: vec![],
+            field_names: None,
+        }
+        .def(rustc_span::DUMMY_SP, self);
+
+        let runtime_array_struct = SpirvType::RuntimeArray {
```

Comment:
My nvidia driver (latest beta 458.17) doesn't seem to be to happy about this one. Using a `[struct; 1]` array instead works, which glslangValidator generates as well

Reply:
I'm not quite sure what this comment refers to and it's not caused any trouble for me so far (I'm on  460.89, but I don't think that should make a huge difference).

Reply:
I tried with the Game Ready Driver 466.11 again and seen same result:

Using a runtime array type results in `0.0` values when loading values through the buffer interface.
spirv-dis:
```
       %uint = OpTypeInt 32 0
%_runtimearr_uint = OpTypeRuntimeArray %uint
 %_struct_20 = OpTypeStruct %_runtimearr_uint
%_runtimearr__struct_20 = OpTypeRuntimeArray %_struct_20
%_ptr_StorageBuffer__runtimearr__struct_20 = OpTypePointer StorageBuffer %_runtimearr__struct_20
         %11 = OpVariable %_ptr_StorageBuffer__runtimearr__struct_20 StorageBuffer
```

Using the mentioned fixed size array correctly loads the values from my shader.
spirv-dis:
```
%uint = OpTypeInt 32 0
%_runtimearr_uint = OpTypeRuntimeArray %uint
 %_struct_20 = OpTypeStruct %_runtimearr_uint
     %uint_1 = OpConstant %uint 1
%_arr__struct_20_uint_1 = OpTypeArray %_struct_20 %uint_1
%_ptr_StorageBuffer__arr__struct_20_uint_1 = OpTypePointer StorageBuffer %_arr__struct_20_uint_1
         %11 = OpVariable %_ptr_StorageBuffer__arr__struct_20_uint_1 StorageBuffer
```

<details><summary>Shader Code</summary>

```rust
#[repr(C)]
#[derive(Copy, Clone)]
pub struct LocalsPbr {
    world_to_view: f32x4x4,
    view_to_clip: f32x4x4,
}

#[spirv(vertex)]
pub fn mesh_vs(
    v_position_obj: f32x3,
    v_normal_obj: f32x3,
    v_texcoord: f32x2,
    v_tangent_obj: f32x4,
    #[spirv(position)] a_position: &mut f32x4,
    a_normal_world: &mut f32x3,
    a_texcoord: &mut f32x2,
    a_tangent_world: &mut f32x4,
    a_position_world: &mut f32x3,
) {
    *a_normal_world = v_normal_obj;
    *a_texcoord = v_texcoord;
    *a_tangent_world = v_tangent_obj;

    let pos_obj = v_position_obj;
    let pos_world = vec4(pos_obj.x, pos_obj.y, pos_obj.z, 1.0);
    *a_position_world = vec3(pos_world.x, pos_world.y, pos_world.z);

    let u_locals = spirv_std::bindless::internal_buffer_load::<LocalsPbr>(0, 0);

    let pos_view = pos_world * u_locals.world_to_view;
    let pos_clip = pos_view * u_locals.view_to_clip;

    *a_position = pos_clip;
}
```
</detail>

Reply:
Ok, so it seems to work when building it as `spirv-unknown-spv1.2` target but not with the `spirv-unknown-spv1.3` or higher target.